### PR TITLE
fix: modernize package

### PR DESCRIPTION
**Pull request by Princesseuh**
This updates the `tsconfig.json` to matches the main repo and its requirements. This unblocks https://github.com/withastro/astro/pull/8519

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,44 +1,43 @@
...
+  "name": "@astrojs/cli-kit",
+  "type": "module",
+  "version": "0.2.5",
+  "license": "MIT",
+  "types": "./dist/index.d.ts",
+  "packageManager": "pnpm@7.11.0",
+  "exports": {
+    ".": {
+			"types": "./dist/index.d.ts",
+      "default": "./dist/index.js"
```
Comment (ematipico) [inline package.json]:
```suggestion
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
```

**Review by ematipico — APPROVED** [review]
Looks good to me, let's ship it! 🚢

File: .changeset/long-icons-warn.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@astrojs/cli-kit": patch
```
Comment (natemoo-re) [inline .changeset/long-icons-warn.md]: I would say this should _at least_ be a `minor`?

**Review by natemoo-re — COMMENTED** [review]
`export` changes are kinda breaking?

**Comment by Princesseuh** [comment]
> `export` changes are kinda breaking?

The exports changes? The only changes to the `exports` field was reversing `types` and `default` since TypeScript want types first

File: .changeset/long-icons-warn.md (hunk 2 above)
Comment (Princesseuh) [reply .changeset/long-icons-warn.md]:
```suggestion
"@astrojs/cli-kit": minor
```

**Pull request merged**